crates/ra_ide_api/src/imp.rs

```diff
@@ -229,19 +231,10 @@ impl db::RootDatabase {
             .collect()
     }
 
-    pub(crate) fn rename(&self, position: FilePosition, new_name: &str) -> Vec<SourceFileEdit> {
-        self.find_all_refs(position)
-            .iter()
-            .map(|(file_id, text_range)| SourceFileEdit {
-                file_id: *file_id,
-                edit: {
-                    let mut builder = ra_text_edit::TextEditBuilder::default();
-                    builder.replace(*text_range, new_name.into());
-                    builder.finish()
-                },
-            })
-            .collect::<Vec<_>>()
+    pub(crate) fn rename(&self, position: FilePosition, new_name: &str) -> Option<SourceChange> {
```

Comment:
We can remove this function now and call `rename::rename(&*self.db)` directly from `Analysis` now.